pkg/nats/nats.go

```diff
@@ -165,12 +165,16 @@ func (n *natsSource) Read(_ context.Context, readRequest sourcesdk.ReadRequest,
 			// Otherwise, we read the data from the source and send the data to the message channel.
 			messageCh <- sourcesdk.NewMessage(
 				[]byte(m.payload),
-				sourcesdk.NewOffset([]byte(m.readOffset), "0"),
+				sourcesdk.NewOffset([]byte(m.readOffset), 0),
 				time.Now())
```

Comment:
Since you are using DefaultPartitions() inside Partitions() method, use the default partition while creating the offset.
```suggestion
				sourcesdk.NewOffsetWithDefaultPartitionId([]byte(m.readOffset)),
				time.Now())
```